src/components/checkoutlogin.jsx

```diff
@@ -0,0 +1,106 @@
+import React, {Component} from 'react';
+import {Form, Button} from 'react-bootstrap';
+
+//Regex to validate the email address.
+const emailRegex = RegExp(
+    /^[a-zA-Z0-9.!#$%&’*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/
+);
+
+//Validate for form errors.
+const formValid = ({ formErrors, ...rest }) => {
+    let valid = true;
+  
+    // validate form errors being empty
+    Object.values(formErrors).forEach(val => {
+      val.length > 0 && (valid = false);
+    });
+  
+    // validate the form was filled out
+    Object.values(rest).forEach(val => {
+      val === null && (valid = false);
+    });
+  
+    return valid;
+};
+
+class CheckoutLogin extends Component {
+
+    constructor(props){
+        super(props);
+        this.state = {
+            email:"",
+            passwrod:"",
+            saveEmail:"",
+        }
+    }
+
+    handleSubmit = e => {
+        e.preventDefault();
+    
+        if (formValid(this.state)) {
+          console.log(`
+            --SUBMITTING--
+            Email: ${this.state.email}
+            Password: ${this.state.password}
+          `);
+        } else {
+          console.error("FORM INVALID - DISPLAY ERROR MESSAGE");
+        }
+    };
+    
+    handleChange = e => {
+        e.preventDefault();
+        const { name, value } = e.target;
+        let formErrors = { ...this.state.formErrors };
+    
+        switch (name) {
+          case "email":
+            formErrors.email = emailRegex.test(value)
+              ? ""
+              : "invalid email address";
+            break;
+          case "password":
+            formErrors.password =
+              value.length < 8? "minimum 8 characaters required" : "";
+            break;
+          default:
+            break;
+        }
+        this.setState({ formErrors, [name]: value }, () => console.log(this.state));
+    };
+
+    render() {
+        const { formErrors } = this.state;
+        
+        return(
+            <div>
```

Comment:
❓ is this `<div>` needed?
❓is the `<div></div>` needed? is it being used as `<br />`?